### ISSUE TITLE
Can't create graph visualizations with inn.SpatialPyramidPooling

### ISSUE THREAD

**Issue by lim0606** [fmassa/optimize-net#9]
Hi, 

First of all, thank you for sharing your works :) I tried to use this package for visualization. It is really cool!

However, I failed to draw the graph of a network architecture with inn.SpatialPyramidPooling Layer. 
(Actually the network is the one in your repository, https://github.com/fmassa/object-detection.torch/blob/refactoring/models/zeiler.lua)

```
/home/jaehyun/github/torch/install/bin/luajit: ...hyun/github/torch/install/share/lua/5.1/nn/Container.lua:69: 
In 3 module of nn.Sequential:
In 1 module of nn.Sequential:
...n/github/torch/install/share/lua/5.1/optnet/graphgen.lua:143: Parent node inexistant for module nn.Linear(12800 -> 4096)
stack traceback:
        [C]: in function 'assert'
        ...n/github/torch/install/share/lua/5.1/optnet/graphgen.lua:143: in function 'addEdge'
        ...n/github/torch/install/share/lua/5.1/optnet/graphgen.lua:168: in function <...n/github/torch/install/share/lua/5.1/optnet/graphgen.lua:162>
        [C]: in function 'xpcall'
        ...hyun/github/torch/install/share/lua/5.1/nn/Container.lua:65: in function 'rethrowErrors'
        ...yun/github/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function <...yun/github/torch/install/share/lua/5.1/nn/Sequential.lua:41>
        [C]: in function 'xpcall'
        ...hyun/github/torch/install/share/lua/5.1/nn/Container.lua:65: in function 'rethrowErrors'
        ...yun/github/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
        ...n/github/torch/install/share/lua/5.1/optnet/graphgen.lua:197: in function 'generateGraph'
        models/create-zf.lua:54: in main chunk
        [C]: in function 'dofile'
        ...thub/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
        [C]: at 0x00406670

WARNING: If you see a stack trace below, it doesn't point to the place where this error occured. Please use only the one above.
stack traceback:
        [C]: in function 'error'
        ...hyun/github/torch/install/share/lua/5.1/nn/Container.lua:69: in function 'rethrowErrors'
        ...yun/github/torch/install/share/lua/5.1/nn/Sequential.lua:44: in function 'forward'
        ...n/github/torch/install/share/lua/5.1/optnet/graphgen.lua:197: in function 'generateGraph'
        models/create-zf.lua:54: in main chunk
        [C]: in function 'dofile'
        ...thub/torch/install/lib/luarocks/rocks/trepl/scm-1/bin/th:145: in main chunk
        [C]: at 0x00406670
```

It seems like that this error would come from the same cause described in https://github.com/fmassa/optimize-net/issues/6. 

Is there anyone who can help me to find this problem? 

Thanks 

Best regards, 

Jaehyun

**Comment by fmassa** [comment]
Hi,

Thanks for opening this issue. There are two possible solutions that I tried:
* change `inn.SpatialPyramidPooling` and `inn.SpatialSameResponseNormalization` to behave like normal modules, and not containers https://github.com/szagoruyko/imagine-nn/commit/ab295ee9d0d2b3843bcba8db0acca0b4c034ddd3
* handle this special case in `optnet` https://github.com/fmassa/optimize-net/commit/541427d136033b9bdca0db7e1ae06981b917d7ce

In the first way, `SpatialPyramidPooling` will be printed as a single module, whereas in the second one it will be printed as an ensemble of modules.

If you update `optimize-net` it should work in the second way I described.

@szagoruyko any thoughts about that ?
